R/github.R

```diff
@@ -181,49 +178,77 @@ use_github_links <- function(auth_token = NULL,
+#'     advice on how to store this.
+#'   * `github_token()` retrieves a stored PAT by consulting, in this order:
+#'     - `GITHUB_PAT` environment variable
+#'     - `GITHUB_TOKEN` environment variable
+#'     - the empty string `""`
 #'
-#' @param scopes Character vector of token permissions. These are just defaults
-#'   that will be pre-selected in the web form. You can select the final values
-#'   on the GitHub page. Read more about GitHub API scopes at
-#'   <https://developer.github.com/apps/building-oauth-apps/scopes-for-oauth-apps/>.
+#' @section: Get and store a PAT:
+#' Sign up for a free [GitHub.com](https://github.com/) account and sign in.
+#' Call `browse_github_token()`. Verify the scopes and click "Generate token".
+#' Copy the token right away! A common approach is to store in `.Renviron` as
+#' the `GITHUB_PAT` environment variable. [edit_r_environ()] opens this file for
+#' editing.
 #'
+#' @param scopes Character vector of token scopes, pre-selected in the web
+#'   form. Final choices are made in the GitHub form. Read more about GitHub
+#'   API scopes at
+#'   <https://developer.github.com/apps/building-oauth-apps/scopes-for-oauth-apps/>.
 #' @param description Short description or nickname for the token. It helps you
 #'   distinguish various tokens on GitHub.
 #' @inheritParams use_github
+#'
+#' @seealso [gh::gh_whoami()] for information on an existing token.
+#'
+#' @return `github_token()` returns a string, a GitHub PAT or `""`.
 #' @export
 #' @examples
 #' \dontrun{
-#' browse_github_pat()
+#' browse_github_token()
 #' ## COPY THE PAT!!!
 #' ## almost certainly to be followed by ...
 #' edit_r_environ()
 #' ## which helps you store the PAT as an env var
 #' }
-browse_github_pat <- function(scopes = c("repo", "gist"),
-                              description = "R:GITHUB_PAT",
-                              host = "https://github.com") {
+browse_github_token <- function(scopes = c("repo", "gist"),
+                                description = "R:GITHUB_PAT",
+                                host = "https://github.com") {
   scopes <- glue_collapse(scopes, ",")
-  url <- sprintf(
-    "%s/settings/tokens/new?scopes=%s&description=%s",
-    host, scopes, description
+  url <- glue(
+    "{host}/settings/tokens/new?scopes={scopes}&description={description}"
   )
   view_url(url)
 
-  ui_todo("Call {ui_code('usethis::edit_r_environ()')} to open {ui_path('.Renviron')}")
+  ui_todo(
+    "Call {ui_code('usethis::edit_r_environ()')} to open {ui_path('.Renviron')}."
+  )
   ui_todo("Store your PAT with a line like:")
   ui_code_block("GITHUB_PAT=xxxyyyzzz")
   ui_todo("Make sure {ui_value('.Renviron')} ends with a newline!")
+  invisible()
+}
+
+#' @rdname browse_github_token
+#' @export
+browse_github_pat <- browse_github_token
+
+#' @rdname browse_github_token
+#' @export
+#' @examples
+#' # for safety's sake, just reveal first 4 characters
+#' substr(github_token(), 1, 4)
+github_token <- function() {
```

Comment:
I wonder if this should print a message (authenticating with GitHub using personal access token) and cleanly error if it can't find a token.

Reply:
I'll look around and see if we are passing/using it in places where it is optional vs. mission critical.

Reply:
This is pretty muddy right now. I will open a separate issue for it because it feel like its own mini-problem.

Reply:
#657 Where to check validity of GitHub PAT?